paper-button.html

```diff
@@ -44,154 +44,131 @@
+<link rel="import" href="../paper-ripple/paper-ripple.html">
+<link rel="import" href="../core-focusable/core-disabled-state.html">
+<link rel="import" href="../core-focusable/core-focused-state.html">
+<link rel="import" href="../core-focusable/core-active-state.html">
+<link rel="import" href="./paper-button-state.html">
 
+<dom-module id="paper-button">
+  <link rel="import" type="css" href="./paper-button.css">
   <template>
+    <!-- Manage disabled -->
+    <core-disabled-state disabled="{{disabled}}"></core-disabled-state>
 
-    <style>
-
-      :host {
-        display: inline-block;
-        position: relative;
-        box-sizing: border-box;
-        min-width: 5.14em;
-        margin: 0 0.29em;
-        background: transparent;
-        text-align: center;
-        font: inherit;
-        text-transform: uppercase;
-        outline: none;
-        border-radius: 3px;
-        -moz-user-select: none;
-        -ms-user-select: none;
-        -webkit-user-select: none;
-        user-select: none;
-        cursor: pointer;
-        z-index: 0;
-      }
+    <!-- Manage focused -->
+    <core-focused-state focused="{{focused}}" disabled="[[disabled]]">
+    </core-focused-state>
 
-      :host([disabled]) {
-        background: #eaeaea;
-        color: #a8a8a8;
-        cursor: auto;
-        pointer-events: none;
-      }
+    <!-- Manage active -->
+    <core-active-state active="{{active}}" disabled="[[disabled]]" toggle="[[toggle]]">
+    </core-active-state>
 
-      ::content * {
-        text-transform: inherit;
-      }
+    <!-- Manage elevation / toggle / raised / aria-pressed -->
+    <paper-button-state
+      elevation="{{elevation}}" toggle="{{toggle}}" raised="{{raised}}"
+      active="[[active]]" focused="[[focused]]" disabled="[[disabled]]">
+    </paper-button-state>
 
-      #bg, #shadow {
-        border-radius: inherit;
-      }
+    <!-- Enable rippliness -->
+    <paper-ripple></paper-ripple>
 
-      #ripple {
-        pointer-events: none;
-        z-index: -1;
-      }
-
-      .button-content {
-        padding: 0.7em 0.57em
-      }
-
-      polyfill-next-selector { content: '.button-content > a'; }
-      ::content > a {
-        height: 100%;
-        padding: 0.7em 0.57em;
-        margin: -0.7em -0.57em;
-        /* flex */
-        -ms-flex: 1 1 0.000000001px;
-        -webkit-flex: 1;
-        flex: 1;
-        -webkit-flex-basis: 0.000000001px;
-        flex-basis: 0.000000001px;
-      }
+    <!-- Reflect elevation -->
+    <paper-card class="content" elevation="[[elevation]]" animated>
 
-    </style>
-
-    <template if="{{raised}}">
-      <paper-shadow id="shadow" fit animated></paper-shadow>
-    </template>
-
-    <!-- this div is needed to position the ripple behind text content -->
-    <div class="button-content" relative layout horizontal center-center>
+      <!-- Re-project light DOM -->
       <content></content>
-    </div>
+    </paper-card>
+  </template>
```

Comment:
I'm a bit concerned that the number of elements in the template causes performance issues.